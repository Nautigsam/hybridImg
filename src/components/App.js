import React, { useState } from 'react';
import { Provider } from 'react-redux';
import { Router, Route } from 'react-router-dom';

import history from '../store/history';
import { store } from '../store';
import ImgGrid from './ImgGrid';
import Navigation from './Navigation';

export function App({ initialData }) {
  return (
    <Router history={history}>
      <Provider store={store}>
        <Navigation />
        <Route
          exact
          path="/mygrid"
          render={() => <ImgGrid edit={true} gridId="G1" />}
        ></Route>
      </Provider>
    </Router>
  );
}
