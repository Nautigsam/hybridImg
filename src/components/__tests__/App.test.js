import React from 'react';
import { StaticRouter } from 'react-router-dom';

import { App } from '../App';

import { shallow } from 'enzyme';

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <StaticRouter>
        <App initialData={{ rawData: {} }} />
      </StaticRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
