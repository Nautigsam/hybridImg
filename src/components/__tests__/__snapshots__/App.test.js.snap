// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <App
    initialData={
      Object {
        "rawData": Object {},
      }
    }
  />
</Router>
`;
